### DISCUSSION TITLE
f-string의 = 포맷팅 기능에 대해

### DISCUSSION THREAD

**Discussion by cbscsm** [8percent/styleguide#21] [General]
릴리즈 노트: https://docs.python.org/3.8/whatsnew/3.8.html#f-strings-support-for-self-documenting-expressions-and-debugging
레퍼런스 문서: https://docs.python.org/3.8/reference/lexical_analysis.html#f-strings
이 기능이 적용된 PR: https://github.com/8percent/eight/pull/12820

위 PR을 적용하기 전에, f-string의 등호 기호 기능에 대한 의견을 나누기 위해 discussion을 생성합니다.

이 기능은 debugging 혹은 logging을 조금 더 편리하게 하기 위해 적용되었습니다.
예시:
```python
a = 'Hello'
b = 'World'
print(f'{a=}')  # a='Hello' <- 기본적으로 repr 적용
print(f'a={a!r}')  # a='Hello'  <- 이전 버전에서 위와 같은 출력 내기
print(f'{a=:*^9}')  # a=**Hello**  <- 포맷이 제공되면 기본적으로 str 적용
print(f'{a=!r:*^9}')  # a=*'Hello'*  <- 포맷도 적용하고, repr도 명시적으로 적용
print(f'{a=!s}')  # a=Hello  <- 명시적으로 str 적용
print(f'{b = }')  # b = 'World'  <- 띄어쓰기 반영됨
print(f'{a+b=}')  # a+b='HelloWorld'  <- 수식도 그대로 찍힘
print(f'{a + b = }')  # a + b = 'HelloWorld'  <- 수식에서도 띄어쓰기 반영
print(f'{", ".join((a, b))=}')  # ", ".join((a, b))='Hello, World'
```
위 PR에서의 예시:
```python
logging.info("Slack message : channel={}, username={}, text={}, attachments={}".format(channel, username, text, attachments))
# 위 코드가 아래 코드로
logging.info(f'Slack message : {channel=!s}, {username=!s}, {text=!s}, {attachments=!s}')
# 그냥 f-string만 적용한다면
logging.info(f'Slack message : channel={channel}, username={username}, text={text}, attachments={attachments}')
# repr 버전끼리 비교하면
logging.info(f'Slack message : {channel=}, {username=}, {text=}, {attachments=}')
logging.info(f'Slack message : channel={channel!r}, username={username!r}, text={text!r}, attachments={attachments!r}')
```

### Cons
- 표현이 다소 직관적이지 않고, 다른 언어와 통용되는 서식이 아닌 점 떄문에 익숙해지기 어려울 수 있습니다.

### Pros
- 그럼에도 적응하고 나면, 여러 로깅/디버깅 혹은 포맷팅 코드에서 중복을 줄이고 간단하게 적을 수 있게 될 것 같습니다.

이러한 표현을 사용하는 것에 대해서 어떻게 생각하시나요?

**Comment by twins1040** [comment]
어떤 장점이 있는지 구체적으로 설명해주시면 좋을 것 같습니다.

**Reply by cbscsm** [reply]
첫 번째/두 번째 예시처럼
```python
print(f'a + b = {a + b}')
```
를
```python
print(f'{a + b = }')
```
이렇게 줄여쓸 수 있습니다.

**Comment by Kirade** [comment]
저는 = 포매팅 기능을 사용하지 않아야 할 이유는 없는것 같습니다.
`{channel=!s}` 과 `channel={channel}` 방식을 혼용해서 사용해볼 수 있을것 같습니다.

**Comment by donghyun0218** [comment]
팀원분들이 새로운 기능의 사용법을 충분히 숙지하고 있다는 가정 하에 새로운 기능을 사용하는 것에 찬성합니다.

**Comment by anohk** [comment]
해당 표현식 사용을 하는것에 찬성합니다. 서로 리뷰 할 때 몇 번 짚어주다 보면 생각보다 금방 적응하지 않을까요?

**Comment by cbscsm** [comment]
#30 PR 생성

**Reply by cbscsm** [reply]
머지완료
